src/main/java/com/mathworks/ci/MatlabTestCase.java

```diff
@@ -0,0 +1,107 @@
+package com.mathworks.ci;
+
+/**
+ * Copyright 2024, The MathWorks Inc.
+ *
+ * Class to store MATLAB test case information
+ * 
+ */
+
+import java.util.List;
+import java.util.ArrayList;
+
+import org.apache.commons.lang.RandomStringUtils;
+
+public class MatlabTestCase {
+    private String name;
+    private List<MatlabTestDiagnostics> diagnostics;
+    private boolean passed;
+    private boolean failed;
+    private boolean incomplete;
+    private String status;
+    private Double duration;
+    private String id;
+
+    public MatlabTestCase() {
+        this.name = "";
+        this.diagnostics = new ArrayList<MatlabTestDiagnostics>();
+        this.passed = false;
+        this.failed = false;
+        this.incomplete = false;
+        this.status = MatlabBuilderConstants.NOT_RUN;
+        this.duration = 0.0;
+        this.id = RandomStringUtils.randomAlphanumeric(8);
+    }
+    
+    public void updateStatus() {
+        if (this.failed){
+            this.status = MatlabBuilderConstants.FAILED;
+        }
+        else if (this.incomplete) {
+            this.status = MatlabBuilderConstants.INCOMPLETE;
+        }
+        else if(this.passed) {
+            this.status = MatlabBuilderConstants.PASSED;
+        }
+    }

```

Comment:
This model of having booleans for `passed`, `failed`, and `incomplete` (which I think might be mutually exclusive in our implementation here but I'm not totally sure) along with a `status` property that seems to have all of that same data wrapped up into one field seems strange to me.

It looks like this `updateStatus` gets called after we set all those 3 other properties, and is derived from the same information, but I'm not sure why we can't just have just one `status` enum property or the booleans.

Reply:
You're actually correct, the status was needed in a previous view in which the info on both the overall test case status as well as Boolean values for each status type was required (for e.g. in case of an assertion failure, the overall test case status would be `failed` but that test case is also marked as `incomplete`). Since we no longer need it, I've removed the Booleans and introduced a `TestStatus` enum.